### DISCUSSION TITLE
models/yolov7/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#17997] [Docs]
# models/yolov7/

Discover YOLOv7, the breakthrough real-time object detector with top speed and accuracy. Learn about key features, usage, and performance metrics.

https://docs.ultralytics.com/models/yolov7/

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for your interest in Ultralytics 🚀! It’s always exciting to see discussions around YOLOv7, a truly groundbreaking model 🎯. For a comprehensive overview of YOLOv7, including key features, usage, and performance metrics, make sure to check out the [YOLOv7 Documentation](https://docs.ultralytics.com/models/yolov7/).

If this is a 🐛 Bug Report, we kindly ask you to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to assist us in correctly identifying and resolving your issue.

If this is a custom training ❓ Question, please share specific details such as datasets, examples of images, training logs, and confirm that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

### Quick Start
YOLOv7 and other YOLO models can be run using one of our verified environments. Use these to explore or train your models with ease:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Docker Image**: <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a> See the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/).

### Install or Upgrade

Ensure you’re using the latest version of the `ultralytics` package, as we’re continuously improving it:

```bash
pip install -U ultralytics
```

This will include all the latest features and fixes! 🌟

### Need More Help?  

Join the growing Ultralytics community to learn and share:  
- 💬 [Discord](https://discord.com/invite/ultralytics)  
- 🌐 [Discourse](https://community.ultralytics.com)  
- 💡 [Reddit](https://reddit.com/r/Ultralytics)  

Rest assured, an Ultralytics engineer will review your discussion soon as well to provide more assistance. 🚀

**Comment by naps22** [comment]
YOLOv7's approach as a trainable bag-of-freebies really stands out, especially in how it optimizes performance without additional inference cost. It's impressive how techniques like re-parameterization and dynamic label assignment enhance accuracy. However, after doing a <a href="https://yolov8.org/yolov8-comparison/">YOLOv8 comparison</a>, it’s clear that while YOLOv8 brings some architectural improvements and streamlined workflows, YOLOv7 still holds strong in many real-time detection scenarios due to its training-time flexibility.

**Reply by glenn-jocher** [reply]
You're absolutely right about YOLOv7's innovative "trainable bag-of-freebies" approach - it was a game-changing concept that improved accuracy purely through training optimizations rather than architectural changes. The re-parameterization and dynamic label assignment techniques were particularly clever solutions that set new standards for efficiency.

While YOLOv7 remains excellent for real-time detection scenarios, you might want to explore [YOLO11](https://docs.ultralytics.com/models/yolo11/) as well - it's our latest model that builds upon these innovations with even better performance across detection, segmentation, and other tasks, all within the streamlined Ultralytics framework.
